TiktokToOSCVRChatGUI.py

```diff
@@ -0,0 +1,161 @@
+import tkinter as tk
+from tkinter import ttk
+from tkinter import scrolledtext
+import asyncio
+import threading
+import logging
+from aioudp import UDPServer
+from TikTokLive import TikTokLiveClient
+from TikTokLive.types.events import GiftEvent, CommentEvent
+import socket
+import os
+
+# Configuration options
+UDP_IP = "127.0.0.1"
+UDP_PORT = 9000
+USERNAME = os.getenv('@username')  # Get username from environment variables
+SLEEP_DURATION = 2  # Duration to sleep after sending a gift or boop
+RETRIES = 5  # Number of retry attempts
```

Comment:
I recommend making it so that constants can be overridden by the environment always, to increase customization.
```
UDP_ID= os.getenv('UDP_IP') or "127.0.0.1"
```